sdk/storage/Azure.Storage.DataMovement/src/DownloadChunkHandler.cs

```diff
@@ -74,6 +75,8 @@ public struct Behaviors
         /// </summary>
         private ConcurrentDictionary<long, string> _rangesCompleted;
 
+        internal ClientDiagnostics ClientDiagnostics { get; }
```

Comment:
Can this one be private?

Reply:
Several probably can but I'm not familiar enough to know which. I'm not sure any of them have a need to be private though. The type is immutable and get-only. Meanwhile the class does expose a getter for its event such that others could raise it. It makes sense to expose the diagnostics necessary for such a raise.